src/raft.c

```diff
@@ -341,24 +361,25 @@ void RaftExecuteCommandArray(RedisRaftCtx *rr,
         int old_entered_eval = rr->entered_eval;
 
         if ((cmdlen == 4 && !strncasecmp(cmd, "eval", 4)) ||
-            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7))) {
+            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7)) ||
+            (cmdlen == 5 && !strncasecmp(cmd, "fcall", 5))) {
             rr->entered_eval = 1;
         }
 
+        /* Explanation:
+         * When we have an ACL, we will have a user set on the context, so need "C"
+         */
+        char *resp_call_fmt = cmds->acl ? "CE0v" : "E0v";
+
         enterRedisModuleCall();
-        reply = RedisModule_Call(ctx, cmd,
-                                 rr->resp_call_fmt, &c->argv[1], c->argc - 1);
-        int ret_errno = errno;
+        RedisModule_SetContextUser(ctx, user);
+        reply = RedisModule_Call(ctx, cmd, resp_call_fmt, &c->argv[1], c->argc - 1);
         exitRedisModuleCall();
 
         rr->entered_eval = old_entered_eval;
 
         if (reply_ctx) {
-            if (reply) {
-                RedisModule_ReplyWithCallReply(reply_ctx, reply);
-            } else {
-                replyRMCallError(reply_ctx, ret_errno, cmd, cmdlen);
-            }
+            RedisModule_ReplyWithCallReply(ctx, reply);
         }
 
         if (reply) {
```

Comment:
```suggestion
```
As @tezc pointed out - we call RM_Call with "E" to reply cannot be NULL.

Reply:
pretty sure this code was already modified?

Reply:
ok, I see, because its consistent now between leader and follower.

Reply:
explanation, please? do we have a scenario when `reply == NULL`?

Reply:
previously we executed RM_Call 2 ways at apply time, with and without E (only reply_ctx node got E).  Ozan argued for it to be the same, so it was simplified.  In the non E (follower/log replay) case, that would have meant errors (say acl) would be reply == NULL (and just ignored as on follower and didnt have an impact).

but I agree now, it can be simplified further